TooltipDecorator/Tooltip.module.less

```diff
@@ -0,0 +1,474 @@
+// Tooltip.module.less
+//
+@import "../styles/variables.less";
+@import "../styles/mixins.less";
+@import "../styles/skin.less";
+
+.applySkins({
+	.tail (@edge, @point) {
+		background-image: linear-gradient(to @edge, @agate-tooltip-border-color @agate-tooltip-border-width, transparent @agate-tooltip-border-width), // Edge line
+		linear-gradient(to @point @edge, @agate-tooltip-bg-color 50%, @agate-tooltip-border-color 50%, @agate-tooltip-border-color ~"calc(50% + 1px)", transparent ~"calc(50% + 1px)"); // Diagonal line + fill
+	}
+	.centerTail (@edge, @point) {
+		background-image: linear-gradient(to @point @edge, @agate-tooltip-bg-color 50%, @agate-tooltip-border-color 50%, @agate-tooltip-border-color ~"calc(50% + 1px)", transparent ~"calc(50% + 1px)"); // Diagonal line + fill
+	}
+
+	.tooltip {
+		position: absolute;
+		z-index: 1;
+
+		// You'll notice tiny adjustments to the position: 0.5px, 1px, 4px, etc. in the following code.
+		// These are to account for sub-pixel rendering on scaled or "imperfect" rendering scenarios
+		// where hairline gaps can be seen between the tail and the body.
+
+		.tooltipArrow {
+			position: absolute;
+			width: @agate-tooltip-point-width;
+			height: @agate-tooltip-point-height;
+			z-index: 1;
+		}
+
+		&.rightArrow,
+		&.leftArrow {
+			.tooltipArrow {
+				width: @agate-tooltip-point-height;
+				height: 12px;
+			}
+		}
+
+		&.centerArrow,
+		&.middleArrow {
+			.tooltipArrow::after {
+				content: "";
+				position: absolute;
+				width: 100%;
+				height: 100%;
+			}
+		}
+
+		&.centerArrow .tooltipArrow {
+			width: ((@agate-tooltip-point-height / 2) + 3px);
+			height: @agate-tooltip-point-width;
+			left: 50%;
+
+			&::after {
+				right: ((@agate-tooltip-point-height / 2) + 1px);
+			}
+		}
+
+		&.middleArrow .tooltipArrow {
+			width: @agate-tooltip-point-width;
+			height: ((@agate-tooltip-point-height / 2) + 2px);
+			top: 50%;
+
+			&::after {
+				bottom: 99%;
+				left: 0;
+			}
+		}
+
+		.tooltipLabel {
```

Comment:
`.moon-small-button-text();` was removed while adapting the moonstone TooltipDecorator to the Agate. Was it intended?

Reply:
Yes, I did not find that necessary.